dd-java-agent/instrumentation/jax-rs-annotations-2/src/main/java/datadog/trace/instrumentation/jaxrs2/JaxRsAnnotationsDecorator.java

```diff
@@ -63,11 +63,10 @@ public void onJaxRsSpan(
     }
   }
 
-  private void updateParent(AgentSpan span, final String resourceName) {
+  private void updateParent(final AgentSpan span, final String resourceName) {
     if (span == null) {
       return;
     }
-    span = span.getLocalRootSpan();
```

Comment:
This is going to cause some weirdness... I couldn't think of a better solution though.  Ideas?

Reply:
I couldn't think of anything better either. I just added this to #50 so we can come back to it later.